### ISSUE TITLE
option for disable strips the underscores

### ISSUE THREAD

**Issue by sky-code** [python-attrs/attrs#391]
Add an option to disable strips the underscores
http://www.attrs.org/en/stable/init.html#private-attributes

I need to create class with `_id` attribute and this is not possible with attrs at this moment

**Comment by hynek** [comment]
You can have an `_id` attribute just fine; it’s just the `__init__` signature is going to be `__init__(_id)`:

```python
In [1]: import attr

In [2]: @attr.s
   ...: class C:
   ...:     _id = attr.ib()
   ...:

In [3]: C(1)
Out[3]: C(_id=1)
```

**Comment by sky-code** [comment]
Problem is in `__init__` method

```python
In [1]: import attr

In [2]: @attr.s
   ...: class C:
   ...:     _id = attr.ib()
   ...:

In [3]: C(**{'_id': 1})
Traceback (most recent call last):
  File "<input>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument '_id'
```
How to add `_id` **as is** without stripping into `__init__` kwargs ?

**Comment by decentral1se** [comment]
`_` stripping should be optional IMHO. I am blocked on adding a `_thing` to my class which also has a `thing` attribute (don't ask, I'm doing a source port). Even if you are right about private attributes "not existing", this choice seems contrary to the stated goals of not "slowing down your code.".

**Comment by decentral1se** [comment]
Furthermore, the documentation states that "There really isn’t a right or wrong, it’s a matter of taste." but one way is not allowed in the software so there is clearly a wrong way to do according to attrs. This is misleading.

**Comment by energizah** [comment]
This issue has come up in some generated code before, for example like this.

```python
@attr.s
class A:
    _ = attr.ib()

File "<attrs generated init __main__.A>", line 2
    self._ =
             ^
SyntaxError: invalid syntax

```

**Comment by csdenboer** [comment]
+1

**Comment by Jasperscheper** [comment]
Any updates on this topic?

**Comment by Tinche** [comment]
A side note, you can use `cattrs` to structure classes with underscore attributes:

```
import attr
import cattr

@attr.s
class C:
  _id = attr.ib()

cattr.structure({'_id': 1}, C)```

**Comment by wbolster** [comment]
closing in favour of #945, which is newer, but has many more details and proposals.
